netket/utils/history.py

```diff
@@ -302,6 +212,59 @@ def __str__(self):
         return f"MVHistory(keys={self.keys()}, n_iters={len(self.iters)})"
 
 
+@dispatch.annotations()
+def append(self: MVHistory, val: MVHistory, it: object = None):
```

Comment:
All of these take `MVHistory` as first argument. Wouldn't it be better to make them class methods then?

Reply:
Not much changes.

Though with this you can define `append` for arbitrary objects to specify how to log them.
If it was a class method you cannot overload it as easily, and to specify how to log a class that already exists you'd need to attach to it a `to_dict` method...

It's just for convenience